decidim-proposals/app/views/decidim/proposals/proposals/_proposal_badge.html.erb

```diff
@@ -1,6 +1,3 @@
-<% if proposal.answered? %>
-  <span class="<%= proposal_state_badge_css_class(proposal.state) %> label proposal-status"><%= humanize_proposal_state proposal.state %></span>
-<% end %>
-<% if proposal.withdrawn? %>
+<% if proposal.published_state? || proposal.withdrawn? %>
```

Comment:
@leio10 is it possible that we required to also check if "Publish proposal answers immediately" is enabled or not?
here and in the show proposal

Reply:
mm, after thinking about this I guess that you are asking me for that change because some Decidim installations are not showing the existing answers after the update. 

Currently, the components that aren't using this feature are marking the answers as published when saved, so there is no need for the change that you requested. As I started using the existing `answered_at` field for this, I didn't find any reason to update the existing proposals. But when I changed that to use the new `state_published_at` field, I didn't remember this decision, and that is what is creating the problem.

My proposal is to add a new data migration to update the `state_published_at` field with the value of the field `answered_at` in all the existing proposals. This should fix the problem and avoid creating additional problems caused by proposals answers that get accidentally published by a change of phase or a setting of the current phase.

What do you think about this? Should I create this new PR?

Reply:
Ok, I like your suggestion. Can you PR to the `release/0.21.0` branch please?